README.md

```diff
@@ -2,4 +2,207 @@
+
+Specifically, the [ocp-diag-core-rust](https://github.com/opencomputeproject/ocp-diag-core-rust) project makes it easy for developers to use the **OCP Test & Validation specification** artifacts by presenting a pure-rust api that can be used to output spec compliant JSON messages.
+
+To start, please see below for [installation instructions](https://github.com/opencomputeproject/ocp-diag-core-rust#installation) and [usage](https://github.com/opencomputeproject/ocp-diag-core-rust#usage).
+
+This project is part of [ocp-diag-core](https://github.com/opencomputeproject/ocp-diag-core) and exists under the same [MIT License Agreement](https://github.com/opencomputeproject/ocp-diag-core-rust/LICENSE).
+
+### Installation
+
+Stable releases of the **ocp-diag-core-rust** codebase are published to **crates.io** under the package name [ocptv](https://crates.io/crates/ocptv), and can be easily installed with rust package manager.
+
+For general usage, the following steps should be sufficient to get the latest stable version using the [Package Installer for Rust](https://github.com/rust-lang/cargo):
+
+- **\[option 1]** using `cargo add`
+
+    ```bash
+    $ cargo add ocptv
+    ```
+
+- **\[option 2]** specifying it in Cargo.toml file
+
+    
+    ```toml
+    [dependencies]
+    ocptv = "1.0.0"
+    ```
+    and then run
+    ```bash
+    $ cargo update
+    ```
+
+To use the bleeding edge instead of the stable version, the git repository should be cloned.
+This assumes that the clone is manually kept up to date by git pulling whenever there are new commits upstream. All of the installation methods below will automatically use the latest library code.
+
+First clone the upstream latest code:
+```bash
+$ git clone https://github.com/opencomputeproject/ocp-diag-core-rust.git
+$ cd ocp-diag-core-rust
+$ git checkout dev # dev branch has the latest code
+```
+Then edit Cargo.toml in your project and add
+
+```
+[dependencies]
+ocptv = { version = "1.0.0", path = "/path/to/ocp-diag-core-rust" }
+```
+
+The instructions above assume a Linux-type system. However, the steps should be identical on Windows and MacOS platforms.
+
+See [The Cargo Book](https://doc.rust-lang.org/cargo/index.html) for more details on how to use cargo.
+
+### Usage
+
+The specification does not impose any particular level of usage. To be compliant, a diagnostic package just needs output the correct artifact messages in the correct format. However, any particular such diagnostic is free to choose what aspects it needs to use/output; eg. a simple validation test may not output any measurements, opting to just have a final Diagnosis outcome.
+
+**Full API reference is available [here](https://docs.rs/ocptv).**
```

Comment:
we should have a section for features; and explain how to use with and without `boxed-scopes` (as in one has the `.scope` method and guarantees end messages, the other the user has to manually write the `end` call)